src/fs/path.rs

```diff
@@ -0,0 +1,240 @@
+//! Module for [Path].
+
+use core::fmt::{Display, Formatter};
+
+/// Separator used for paths accross multiple directories. Also used as identifier for the
+/// root path.
+pub const SEPARATOR: char = '/';
+/// [SEPARATOR] but as string.
+const SEPARATOR_STR: &str = "/";
+/// Each component in FAT-like file systems has no more bytes longer than this.
+/// Length without NULL byte.
+pub const MAX_COMPONENT_LENGTH: usize = 255;
+
+/// Deny list of characters for path components.
+/// UEFI supports FAT-like file systems. According to <https://en.wikipedia.org/wiki/Comparison_of_file_systems>
+/// paths should not contain the following symbols.
+pub const CHARACTER_DENY_LIST: [char; 10] =
+    ['\0', '"', '*', SEPARATOR, ':', '<', '>', '?', '\\', '|'];
+
+/// Errors that can happen during [PathBuf] creation.
+#[derive(Debug, Clone, PartialOrd, PartialEq, Hash, Ord, Eq)]
+pub enum PathError {
+    /// The path contains at least one component that contains illegal characters.
+    /// See [CHARACTER_DENY_LIST] for denied characters.
+    IllegalComponent,
+    /// The path contains at least one empty component. This may happen from `/a///b`.
+    EmptyComponents,
+    /// The path is entirely empty.
+    EmptyPath,
+}
+
+/// A path is a small abstraction over a [str]. It is used to validate a path and iterate over its
+/// components. A path is valid inside a corresponding [super::FileSystem]. There is no support for
+/// ".." and ".". All paths start from the root of a [super::FileSystem]. However, the leading
+/// "/" is not required. Empty paths, i.e., "" and "/", are denied.
+#[derive(Debug, Default, Hash, PartialOrd, PartialEq, Ord, Eq)]
+pub struct Path<'a>(&'a str);
```

Comment:
I wonder if there's any existing `no_std` library on crates.io that we could use for path stuff? I didn't find one from a very cursory search, but might have missed something.

Reply:
I also didn't find any. As the `path` module is rather simple, I do not see a big risk of providing the code by ourselves. Additionally, there are tests that cover corner cases and the happy path. Is this okay?

Reply:
Yep seems fine, just thought it was worth taking an extra look.